oteapi_dlite/strategies/add_instance.py

```diff
@@ -0,0 +1,98 @@
+"""Generic function strategy using DLite storage plugin."""
+# pylint: disable=unused-argument,invalid-name
+from typing import TYPE_CHECKING, Optional
+
+import dlite
+from dlite.utils import infer_dimensions
+from oteapi.models import AttrDict, FunctionConfig, SessionUpdate
+from pydantic import Field
+from pydantic.dataclasses import dataclass
+
+from oteapi_dlite.models import DLiteSessionUpdate
+from oteapi_dlite.utils import get_collection, update_collection
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+
+    from oteapi.interfaces import IFunctionStrategy
+
+
+class AddInstanceConfig(AttrDict):
+    """Configuration for adding an instance to the collection."""
+
+    datamodel: str = Field(
+        description="ID (URI or UUID) of the datamodel.",
```

Comment:
It should be mentioned that the data model must be available in `dlite.storage_path`.